packages/lix-sdk/src/query-utilities/get-lowest-common-ancestor-v2.ts

```diff
@@ -0,0 +1,110 @@
+import type { Change } from "../database/schema.js";
+import type { LixReadonly } from "../plugin/lix-plugin.js";
+
+/**
+ * Returns the lowest common ancestor of two changes.
+ *
+ * @returns Either the lowest common ancestor of the two changes or `undefined` if they do not share a common ancestor.
+ */
+export async function getLowestCommonAncestorV2(args: {
+	lix: Pick<LixReadonly, "db">;
+	changeA: Pick<Change, "id">;
+	changeB: Pick<Change, "id">;
+}): Promise<Change | undefined> {
+	// check for direct ancestors first before running the recursive query
+	const directAncestorA = await args.lix.db
```

Comment:
This is (ofc) a monster query. Performance is likely sub-optimal. Doesn't matter for now. We can add caching or jumping of commonly/previously traversed paths in the future

Reply:
🤯 looking at this for a while now :D

Reply:
A thought here: coverage wont be a good measurement for tests of lix i guess...